[MRG] Add examples for vectorizer classes

mepa opened the pull request:
#### Reference Issues/PRs
See [#3846](https://github.com/scikit-learn/scikit-learn/issues/3846)

#### What does this implement/fix? Explain your changes.
This PR adds examples for the following classes:
- HashingVectorizer
- CountVectorizer
- TfidfVectorizer

mepa committed:
Add examples for vectorizer classes
- HashingVectorizer
- CountVectorizer
- TfidfVectorizer

amueller commented on sklearn/feature_extraction/text.py:
```diff
@@ -479,6 +479,46 @@ class HashingVectorizer(BaseEstimator, VectorizerMixin, TransformerMixin):
+     [ 0.  0.  0. ...,  0.  0.  0.]]
+    >>> print(X.shape)
+    (4, 1048576)
+    >>>
```
I would remove this and the other empty line above.

amueller reviewed (approved):
Looks good if the tests pass. Ignore LGTM

mepa committed:
Remove empty lines

qinhanmin2014 commented on sklearn/feature_extraction/text.py:
```diff
@@ -479,6 +479,44 @@ class HashingVectorizer(BaseEstimator, VectorizerMixin, TransformerMixin):
+    >>> from sklearn.feature_extraction.text import CountVectorizer
+    >>> corpus = [
+    ...     'This is the first document.',
+    ...     'This is the second second document.',
```
Maybe we should provide users with a meaningful sentence, so remove one ``second``?

mepa replied:
Good point. I'll change the sentence to be more meaningful.

qinhanmin2014 commented on sklearn/feature_extraction/text.py:
```diff
@@ -479,6 +479,44 @@ class HashingVectorizer(BaseEstimator, VectorizerMixin, TransformerMixin):
+     [ 0.  0.  0. ...,  0.  0.  0.]]
+    >>> print(X.shape)
+    (4, 1048576)
+    >>> vectorizer = HashingVectorizer(n_features=2**3)
```
There're actually two examples here. We aim to provide users with a minimal example and your examples seems too long from my side. Maybe remove the previous example, unless you have some specific reason?

mepa replied:
I'll remove the first example with default `n_features=2**20`, plus shorten it.

mepa committed:
Provide more meaningful sentences

Also, remove part of the HashingVectorizer example with default
n_features = 2**20. Use only the second example with
n_features = 2**4, and shorten.

rth reviewed (commented):
The rendering in CountVectorizer has an issue see https://27800-843222-gh.circle-artifacts.com/0/doc/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html

mepa committed:
Avoid printing arrays with values of numerical precision 8

The doctest fails for precision > 7. It's illustrative in this case
to show the array shape rather than adding extra lines to set the
print precision to <= 7.

glemaitre commented on sklearn/feature_extraction/text.py:
```diff
@@ -746,6 +784,27 @@ class CountVectorizer(BaseEstimator, VectorizerMixin):
 
         This is only available if no vocabulary was given.
 
+    >>> from sklearn.feature_extraction.text import TfidfVectorizer
```
You are missing 

```python
Examples
--------
```

mepa replied:
Fixed

glemaitre commented on sklearn/feature_extraction/text.py:
```diff
@@ -1377,6 +1436,25 @@ class TfidfVectorizer(CountVectorizer):
+    ...     'And the third one.',
+    ...     'Is this the first document?',
+    ... ]
+    >>> vectorizer = CountVectorizer()
```
I would expect to have a TfidfVectorizer in this docstring, isn't it?

mepa replied:
Same as above: [commit](https://github.com/scikit-learn/scikit-learn/pull/11534/commits/f7fd421f3454aaee9fe4d8f88e8258e36b8b7ca3)

glemaitre commented on sklearn/feature_extraction/text.py:
```diff
@@ -746,6 +784,27 @@ class CountVectorizer(BaseEstimator, VectorizerMixin):
 
         This is only available if no vocabulary was given.
 
+    >>> from sklearn.feature_extraction.text import TfidfVectorizer
```
We should illustrate only the CountVectorizer here

mepa replied:
Fixed with [this commit](https://github.com/scikit-learn/scikit-learn/pull/11534/commits/f7fd421f3454aaee9fe4d8f88e8258e36b8b7ca3).

mepa commented:
Hi @glemaitre. You're right, the examples for TfidfVectorizer and CountVectorizer are mixed. I'll fix.

mepa committed:
Fix mixed examples for tfidf and count vectorizers

qinhanmin2014 reviewed (approved):
LGTM, thanks @mepa

mepa commented:
Thanks @qinhanmin2014
